src/libCom/test/epicsTimerTest.cpp

```diff
@@ -27,6 +35,150 @@
 #define verify(exp) ((exp) ? (void)0 : \
     epicsAssert(__FILE__, __LINE__, #exp, epicsAssertAuthor))
 
+#if __cplusplus >= 201103L
+const double timeTolerance { 1e-3 + epicsThreadSleepQuantum() };
+
+class handler : public epicsTimerNotify
+{
+public:
+  handler(epicsTimerQueue& queue, std::function<void()> expireFctIn = [] {}) : timer(queue.createTimer()), expireFct(expireFctIn) {}
+  ~handler() { timer.destroy(); }
+  void start(double delay) {
+    startTime = epicsTime::getCurrent();
+    timer.start(*this, delay);
+  }
+  void cancel() { timer.cancel(); }
+  expireStatus expire(const epicsTime& currentTime) {
+    expireTime = epicsTime::getCurrent();
+    expireCount++;
+    expireFct();
+    return expireStatus(noRestart);
+  }
+  int getExpireCount() const { return expireCount; }
+  double getDelay() const { return expireTime - startTime; }
+private:
+  epicsTimer &timer;
+  int expireCount { 0 };
+  epicsTime startTime;
+  epicsTime expireTime;
+  std::function<void()> expireFct;
+};
+
+void verifyExpirationTime(double delta_t, double tolerance) {
+  if (!testOk(std::abs(delta_t) < tolerance, "timer expired at the expected "
+              "time (error = %f ms, tolerance = +/-%f ms)", 1000.0 * delta_t,
+              1000.0 * tolerance)) {
+    const char * msg = delta_t < 0.0 ? "early" : "late";
+    testDiag("delta t = %g ms (timer expired too %s)", 1000.0 * delta_t, msg);
+  }
+}
+
+void testTimerExpires() {
+  testDiag("Timer expires");
+  epicsTimerQueueActive &queue = epicsTimerQueueActive::allocate(false);
+  {
+    handler h1(queue);
+    const double arbitrary_time { 0.3 };
+    h1.start(arbitrary_time);
+
+    epicsThreadSleep(arbitrary_time + 0.1);
```

Comment:
I'm not enthused by all of the sleeps, especially with hard-coded delays.  I would like to see all of these tests using `epicsEvent`.

Reply:
Seems like I forgot to commit my latest changes. Eliminating the first few `epicsThreadSleep`s was easy but I don't have any ideas how to get rid of the ones related to the cancel tests.